### DISCUSSION TITLE
Microsoft Devcontainer Bump

### DISCUSSION THREAD

**Discussion by rfpludwick** [renovatebot/renovate#33260] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub & Gitea, Always Latest Version

### Please tell us more about your question or problem

I have a Dockerfile which I use to build my own flavor of devcontainer. Renovate seems unable to determine that there are newer versions of the devcontainer available in the registry. Pertinent Dockerifle snippet:

```
FROM mcr.microsoft.com/vscode/devcontainers/base:ubuntu-22.04
```

I'm unsure how to resolve this so Renovate will correctly see that `mcr.microsoft.com/vscode/devcontainers/base:ubuntu-24.04` exists.

### Logs (if relevant)

<details><summary>Logs</summary>

```

       "config": {
         "dockerfile": [
           {
             "deps": [
               {
                 "depName": "mcr.microsoft.com/vscode/devcontainers/base",
                 "currentValue": "ubuntu-22.04",
                 "replaceString": "mcr.microsoft.com/vscode/devcontainers/base:ubuntu-22.04",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "final",
                 "updates": [],
                 "packageName": "mcr.microsoft.com/vscode/devcontainers/base",
                 "versioning": "docker",
                 "warnings": [],
                 "skipReason": "invalid-value"
               }
             ],
             "packageFile": "Dockerfile"
           }
         ]
       }

```

</details>

**Comment by viceice** [comment, marked as answer]
you need to override versioning.

- use regex versioning 
- use Ubuntu versioning with version compatibility

**Reply by rfpludwick** [reply]
Thanks, that did the trick. I forgot all about having to put the `compatibility` into a regex; I've done that elsewhere. For those who want to know the configuration I added:

```json5
packageRules: [
	{
		matchDatasources: [
			"docker",
		],
		matchPackageNames: [
			"mcr.microsoft.com/vscode/devcontainers/base",
		],
		versioning: "regex:^(?<compatibility>[^-]+)-(?<major>\\d+)\\.(?<minor>\\d+)$",
	},
],
```

**Reply by viceice** [reply]
version compatibility with Ubuntu versioning is probably a better solution 

https://docs.renovatebot.com/configuration-options/#versioncompatibility
